File: codegen-tool/src/main/java/com/dqjq/codegen/Generator/ServiceGenerator.java
package com.dqjq.codegen.Generator;

import com.dqjq.codegen.CodegenConfig;
import lombok.Getter;

import java.io.IOException;
import java.nio.file.Path;

/**
 * Service 接口生成器
 * <p>
 * 负责生成业务逻辑接口。
 * </p>
 */
@Getter
public class ServiceGenerator extends BaseGenerator {

    private Path servicePath;

    public ServiceGenerator(CodegenConfig cfg) {
        super(cfg);
    }

    @Override
    public void generate() throws IOException {
        Path baseOut = getBaseOutputDir();
        String basePackagePath = cfg.packageBase.replace('.', '/');
        String modulePath = baseOut + "/" + basePackagePath + "/" + cfg.module;
        Path serviceDir = projectRoot.resolve(modulePath + "/service");
        
        ensureDirectory(serviceDir);
        
        this.servicePath = serviceDir.resolve(cfg.entityName + "Service.java");

        String template = readTemplate("service.tpl");
        if (template.isEmpty()) {
            throw new IOException("无法加载 Service 模板: service.tpl");
        }

        String servicePkg = cfg.packageBase + "." + cfg.module + ".service";
        String dtoPkg = cfg.packageBase + "." + cfg.module + ".dto";
        String entityPkg = cfg.packageBase + "." + cfg.module + ".entity";

        String content = template
                .replace("{{package}}", servicePkg)
                .replace("{{dtoPackage}}", dtoPkg)
                .replace("{{entityPackage}}", entityPkg)
                .replace("{{entityName}}", cfg.entityName)
                .replace("{{dtoName}}", cfg.entityName + "Dto")
                .replace("{{idType}}", cfg.idType);

        writeFile(servicePath, content);
    }

}

File: codegen/codegen-tool-py/codegen/generators/service_generator.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Service生成器
生成Java Service接口
"""

from pathlib import Path
from codegen.generators.base_generator import BaseGenerator

class ServiceGenerator(BaseGenerator):
    """Service生成器"""
    
    def generate(self):
        """
        生成Service接口
        """
        # 准备上下文数据
        context = {
            'package': self.get_full_package('service'),
            'entity_name': self.code_generator.entity_name,
            'id_type': self.code_generator.id_type,
            'entity_package': self.get_full_package('entity'),
            'dto_package': self.get_full_package('dto'),
            'service_base_class': self.config_dict.get('serviceBaseClass', ''),
            'has_base_class': bool(self.config_dict.get('serviceBaseClass', ''))
        }
        
        # 渲染模板
        content = self.render_template('java/service.j2', context)
        
        # 生成文件路径
        output_dir = self.code_generator.get_output_path('service')
        self.output_path = output_dir / f"{self.code_generator.entity_name}Service.java"
        
        # 写入文件
        self.write_file(self.output_path, content)
